File: union_find/quick_find.py
import unittest

class QuickFindUF:

    def __init__(self, n):
      self.n = n
      self.array = list(range(n))

    def connected(self, x, y):
      return self.array[x] == y

    def union(self, x, y):
      for i in range(self.n):
        if self.array[i] == x:
          self.array[i] = y

class QuickFindUFTest(unittest.TestCase):
  
    def setUp(self):
        self.obj = QuickFindUF(10)

    def test_initialize(self):
        expected = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
        self.assertEqual(self.obj.array, expected)

    def test_connected(self):
        self.assertFalse(self.obj.connected(2, 3))

    def test_union(self):
        self.obj.union(2, 3)
        self.assertTrue(self.obj.connected(2, 3))

if __name__ == '__main__':
    unittest.main()

File: 01_union_find/quick_find.rb
require 'test/unit'

class QuickFindUF
 
 attr_reader :array

 def initialize n
  @n = n-1
  @array = (0..@n).to_a
 end

 def connected x, y
   @array[x] == y
 end

 def union x, y
  (0..@n).each do |i|
    @array[i] = y if @array[i] == x
  end
 end

end


class QuickFindUFTest < Test::Unit::TestCase
  
  def setup
    @obj = QuickFindUF.new(10)
  end

  def test_initialize
    expected = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
    assert_equal(expected, @obj.array)
  end

  def test_connected
    assert_equal(false, @obj.connected(2, 3))
  end

  def test_union
    @obj.union(2, 3)
    assert_equal(true, @obj.connected(2, 3))
  end

end